bids/variables/io.py

```diff
@@ -345,8 +345,10 @@ def _load_tsv_variables(layout, type_, dataset=None, columns=None,
         # Filter rows on all selectors
         comm_cols = list(set(_data.columns) & set(selectors.keys()))
         for col in comm_cols:
-            vals = listify(selectors.get(col))
-            _data = _data.query('%s in @vals' % col)
+            filter = [False] * _data.shape[0]
+            for val in listify(selectors.get(col)):
+                filter = filter | _data[col].str.match(val)
```

Comment:
I'm also not understanding what this line is supposed to do. What is the type of `_data[col].str.match(val)` and what how does a binary or with a list of bools work?

Reply:
`str.match` returns a pandas Series of bools, which is true for matching rows. The `or` is elementwise, so as it iterates over the possible selectors, it will continue to add any matching rows, and then the dataframe is subset once. 

I'm not sure why I ended up doing it that way, rather than just taking the set of matching columns, but I did.